pallets/xcm/src/lib.rs

```diff
@@ -190,3 +191,14 @@ where
 		_ => Err(BadOrigin),
 	}
 }
+
+/// Accepts an asset if it is a native asset of a particular type.
+pub struct ConcreteNativeAsset<AssetId>(PhantomData<AssetId>);
```

Comment:
```suggestion
/// Accepts an asset if it is a native asset from a particular `MultiLocation`.
pub struct ConcreteNativeAssetFrom<Location>(PhantomData<AssetId>);
```

Reply:
Also, what was the reason to put this struct inside of the XCM pallet?

Reply:
For potential reuse across parachain runtimes, it also has dependencies on xcm, so that makes sense. As discussed with @joepetrowski

Reply:
No it doesn't. `DenyThenTry` lives in `parachains/common/src/xcm_config.rs`; why isn't this struct also living there instead?

Reply:
Yeah, this should definitely go where Keith suggested, as part of the parachains' runtime config

Reply:
Will fix that